Backend Kubernetes Panic when no WOODPECKER_BACKEN_K8S_POD_LABELS are provided

### Component

agent

### Describe the bug

When running an agent with the Kubernetes backend, not setting any labels in the agent config (`WOODPECKER_BACKEN_K8S_POD_LABELS`) causes an assignment to a non initialized string map.


```text
panic: assignment to entry in nil map

goroutine 90 [running]:
github.com/woodpecker-ci/woodpecker/pipeline/backend/kubernetes.Pod({0xc000054051, 0xa}, 0xc00016ef00, 0x0, 0x0)
	/src/pipeline/backend/kubernetes/pod.go:91 +0x81b
github.com/woodpecker-ci/woodpecker/pipeline/backend/kubernetes.(*kube).Exec(0xc0002819e0, {0x1cf52c0, 0xc0003ea440}, 0x2900450?)
	/src/pipeline/backend/kubernetes/kubernetes.go:165 +0x65
github.com/woodpecker-ci/woodpecker/pipeline.(*Runtime).exec(0xc00046c540, 0xc00016ef00)
	/src/pipeline/pipeline.go:218 +0x42
github.com/woodpecker-ci/woodpecker/pipeline.(*Runtime).execAll.func1()
	/src/pipeline/pipeline.go:186 +0x1ce
golang.org/x/sync/errgroup.(*Group).Go.func1()
	/src/vendor/golang.org/x/sync/errgroup/errgroup.go:75 +0x64
created by golang.org/x/sync/errgroup.(*Group).Go
	/src/vendor/golang.org/x/sync/errgroup/errgroup.go:72 +0xa5
{"level":"info","time":"2023-01-04T01:00:43Z","message":"Starting Woodpecker agent with version 'next-f1e9c841' and backend 'kubernetes' running up to 1 pipelines in parallel"}
```

### System Info

```shell
Server: next-f1e9c841
Agent next-f1e9c841 and backend kubernetes
```


### Additional context

The Issue was likely introduced in #1510 since it completely changed the label handling (before that there was just a single statement initializing and setting the step label).

The line causing the panic is the assignment of the `labels["step"] = podName(step)`, if there is no extra label provided in the agents config, `labels` is uninitialized at this point (nil), due to the `yaml.Unmarshal` not writing anything into the `&config.PodLabels` which causes the error.

Unmarshalling

https://github.com/woodpecker-ci/woodpecker/blob/f1e9c8416dafb6c236694765d9f7dde9620592da/pipeline/backend/kubernetes/kubernetes.go#L61

Assignment where the assignment to nil happens

https://github.com/woodpecker-ci/woodpecker/blob/f1e9c8416dafb6c236694765d9f7dde9620592da/pipeline/backend/kubernetes/pod.go#L91

Just noting this down here before I have time to fix it...

I'd tend to fix this by doing a nil check after the unmarshaling of the podLabels in kubernetes.go#61 (before the return), any opinions on that or should it rather be fixed by checking for nil in within the function in pod.go, before reaching pod.go#91 ?




### Validations

- [X] Read the [Contributing Guidelines](https://github.com/woodpecker-ci/woodpecker/blob/master/CONTRIBUTING.md).
- [X] Read the [docs](https://woodpecker-ci.org/docs/intro).
- [X] Check that there isn't [already an issue](https://github.com/woodpecker-ci/woodpecker/issues) that reports the same bug to avoid creating a duplicate.
- [X] Check that this is a concrete bug. For Q&A join our [Discord Chat Server](https://discord.gg/fcMQqSMXJy) or the [Matrix room](https://matrix.to/#/#woodpecker:matrix.org).

--- a/pipeline/backend/kubernetes/kubernetes.go
+++ b/pipeline/backend/kubernetes/kubernetes.go
@@ -58,15 +58,17 @@ func configFromCliContext(ctx context.Context) (*Config, error) {
 				StorageRwx:   c.Bool("backend-k8s-storage-rwx"),
 			}
 			// Unmarshal label and annotation settings here to ensure they're valid on startup
-			err := yaml.Unmarshal([]byte(c.String("backend-k8s-pod-labels")), &config.PodLabels)
-			if err != nil {
-				log.Error().Msgf("could not unmarshal pod labels '%s': %s", c.String("backend-k8s-pod-labels"), err)
-				return nil, err
+			if labels := c.String("backend-k8s-pod-labels"); labels != "" {
+				if err := yaml.Unmarshal([]byte(labels), &config.PodLabels); err != nil {
+					log.Error().Msgf("could not unmarshal pod labels '%s': %s", c.String("backend-k8s-pod-labels"), err)
+					return nil, err
+				}
 			}
-			err = yaml.Unmarshal([]byte(c.String("backend-k8s-pod-annotations")), &config.PodAnnotations)
-			if err != nil {
-				log.Error().Msgf("could not unmarshal pod annotations '%s': %s", c.String("backend-k8s-pod-annotations"), err)
-				return nil, err
+			if annotations := c.String("backend-k8s-pod-annotations"); annotations != "" {
+				if err := yaml.Unmarshal([]byte(c.String("backend-k8s-pod-annotations")), &config.PodAnnotations); err != nil {
+					log.Error().Msgf("could not unmarshal pod annotations '%s': %s", c.String("backend-k8s-pod-annotations"), err)
+					return nil, err
+				}
 			}
 			return &config, nil
 		}

--- a/pipeline/backend/kubernetes/kubernetes.go
+++ b/pipeline/backend/kubernetes/kubernetes.go
@@ -43,9 +43,9 @@ type Config struct {
 	Namespace      string
 	StorageClass   string
 	VolumeSize     string
+	StorageRwx     bool
 	PodLabels      map[string]string
 	PodAnnotations map[string]string
-	StorageRwx     bool
 }
 
 func configFromCliContext(ctx context.Context) (*Config, error) {

--- a/pipeline/backend/kubernetes/kubernetes.go
+++ b/pipeline/backend/kubernetes/kubernetes.go
@@ -52,10 +52,12 @@ func configFromCliContext(ctx context.Context) (*Config, error) {
 	if ctx != nil {
 		if c, ok := ctx.Value(types.CliContext).(*cli.Context); ok {
 			config := Config{
-				Namespace:    c.String("backend-k8s-namespace"),
-				StorageClass: c.String("backend-k8s-storage-class"),
-				VolumeSize:   c.String("backend-k8s-volume-size"),
-				StorageRwx:   c.Bool("backend-k8s-storage-rwx"),
+				Namespace:      c.String("backend-k8s-namespace"),
+				StorageClass:   c.String("backend-k8s-storage-class"),
+				VolumeSize:     c.String("backend-k8s-volume-size"),
+				StorageRwx:     c.Bool("backend-k8s-storage-rwx"),
+				PodLabels:      make(map[string]string), // just init empty map to prevent nil panic
+				PodAnnotations: make(map[string]string), // just init empty map to prevent nil panic
 			}
 			// Unmarshal label and annotation settings here to ensure they're valid on startup
 			if labels := c.String("backend-k8s-pod-labels"); labels != "" {

